store/useStore.js

```diff
@@ -115,48 +85,85 @@ const useStore = create((set) => {
               ],
             },
           },
-          ...state.games,
-        ];
-        return {
-          games: newGameList,
-        };
-      });
-    },
-
-    // delete card with "x"-icon in gamelist
-    deleteGame: (id) => {
-      set((state) => {
-        const gameListAfterDeletion = state.games.filter((game) => {
-          return game.id !== id;
-        });
-        return {
-          games: gameListAfterDeletion,
-        };
-      });
-    },
+        ],
 
-    // used https://www.robinwieruch.de/react-update-item-in-list/ as tutorial
-    updateGameDetail: (id, gameresult, set1, set2, set3) => {
-      set((state) => {
-        const updatedGameList = state.games.map((game) => {
-          if (game.id === id) {
-            const gameToUpdate = {
-              ...game,
-              results: {
-                gameresult: gameresult,
-                set: [set1, set2, set3],
+        // append new card via form
+        appendNewGame: (type, name, date, time, place, court) => {
+          set((state) => {
+            const newGameList = [
+              {
+                id: nanoid(),
+                type: type,
+                name: name,
+                date: date,
+                time: time,
+                place: place,
+                court: court,
+                results: {
+                  gameresult: undefined,
+                  set: [
+                    {
+                      Player1: "",
+                      Player2: "",
+                    },
+                    {
+                      Player1: "",
+                      Player2: "",
+                    },
+                    {
+                      Player1: "",
+                      Player2: "",
+                    },
+                  ],
+                },
               },
+              ...state.games,
+            ];
+            return {
+              games: newGameList,
             };
-            return gameToUpdate;
-          }
-          return game;
-        });
-        return {
-          games: updatedGameList,
-        };
-      });
+          });
+        },
+
+        // delete card with "x"-icon in gamelist
+        deleteGame: (id) => {
+          set((state) => {
+            const gameListAfterDeletion = state.games.filter((game) => {
+              return game.id !== id;
+            });
```

Comment:
u could shorten this by writing
`const gameListAfterDeletion = state.games.filter((game) => game.id !== id;`